tests/dogfood.rs

```diff
@@ -41,12 +41,77 @@ fn dogfood_clippy() {
 
 #[test]
 fn dogfood_subprojects() {
+    fn test_no_deps_ignores_path_deps_in_workspaces() {
+        fn clean(cwd: &Path, target_dir: &Path) {
+            Command::new("cargo")
+                .current_dir(cwd)
+                .env("CARGO_TARGET_DIR", target_dir)
+                .arg("clean")
+                .args(&["-p", "subcrate"])
+                .args(&["-p", "path_dep"])
+                .output()
+                .unwrap();
+        }
+
+        if cargo::is_rustc_test_suite() {
+            return;
+        }
+        let root = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let target_dir = root.join("target").join("dogfood");
+        let cwd = root.join("clippy_workspace_tests");
+
+        // Make sure we start with a clean state
+        clean(&cwd, &target_dir);
```

Comment:
I don't like this. IIUC this test is run after the dogfood tests are run, right? This will delete the results from the dogfood run, and when you run dogfood again, you have to recompile everything. That way rerunning dogfood takes ~1-3min, if we wouldn't clean it it only takes a few seconds to rerun dogfood.

Haven't tested if what I wrote above is true, though :D

---

Why not a new target dir for this? That target dir should be really small anyway.

Reply:
We are passing the subcrates to the `clean` command, so only those are cleaned. I just tested it and re-running the dogfood for subprojects is almost instantaneous in my machine.

If I'm not mistaken, the `clean` call is necessary because of the problem I mentioned in [another comment](https://github.com/rust-lang/rust-clippy/pull/6188#issuecomment-734901249) (no rebuilds when flags change), so we should be able to remove those calls soon.

If you still think using another target dir is necessary I will of course look into it!

Reply:
Thanks for testing this! In that case nothing has to be done here.